src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_FixedStatement.cs

```diff
@@ -286,6 +290,106 @@ public override BoundNode VisitFixedLocalCollectionInitializer(BoundFixedLocalCo
             return factory.Block(pinnedTempInit, localInit);
         }
 
+        /// <summary>
+        /// fixed(int* ptr = &amp;v){ ... }    == becomes ===>
+        /// 
+        /// pinned ref int pinnedTemp = ref v;    // pinning managed ref
+        /// int* ptr = (int*)&amp;pinnedTemp;         // unsafe cast to unmanaged ptr
+        ///   . . . 
+        /// </summary>
+        private BoundStatement InitializeFixedStatementGetPinnable(
+            BoundLocalDeclaration localDecl,
+            LocalSymbol localSymbol,
+            BoundFixedLocalCollectionInitializer fixedInitializer,
+            SyntheticBoundNodeFactory factory,
+            out LocalSymbol pinnedTemp)
+        {
+            TypeSymbol localType = localSymbol.Type;
+            BoundExpression initializerExpr = VisitExpression(fixedInitializer.Expression);
+
+            var initializerType = initializerExpr.Type;
+            var initializerSyntax = initializerExpr.Syntax;
+            var getPinnableMethod = fixedInitializer.GetPinnableOpt;
+
+            // intervening parens may have been skipped by the binder; find the declarator
+            VariableDeclaratorSyntax declarator = fixedInitializer.Syntax.FirstAncestorOrSelf<VariableDeclaratorSyntax>();
+            Debug.Assert(declarator != null);
+
+            pinnedTemp = factory.SynthesizedLocal(
+                getPinnableMethod.ReturnType,
+                syntax: declarator,
+                isPinned: true,
+                //NOTE: different from the array and string cases
+                //      RefReadOnly to allow referring to readonly variables. (technically we only "read" through the temp anyways)
+                refKind: RefKind.RefReadOnly,
+                kind: SynthesizedLocalKind.FixedReference);
+
+            BoundExpression callReceiver;
+            int currentConditionalAccessID = 0;
+
+            bool needNullCheck = !initializerType.IsValueType;
+
+            if (needNullCheck)
+            {
+                currentConditionalAccessID = _currentConditionalAccessID++;
+                callReceiver = new BoundConditionalReceiver(
+                    initializerSyntax,
+                    currentConditionalAccessID,
+                    initializerType);
+            }
+            else
+            {
+                callReceiver = initializerExpr;
+            }
+
+            // .GetPinnable()
+            var getPinnableCall = getPinnableMethod.IsStatic?
+                                            factory.Call(null, getPinnableMethod, callReceiver):
+                                            factory.Call(callReceiver, getPinnableMethod);
+
+            // temp =ref .GetPinnable()
+            var tempAssignment = factory.AssignmentExpression(
+                factory.Local(pinnedTemp),                                                   
+                getPinnableCall,
+                isRef: true);
+
+            // &pinnedTemp;
```

Comment:
No semi-colon

Reply:
Nit: doesn't need a semi-colon: `// &pinnedTemp`

---
In reply to: [167677135](https://github.com/dotnet/roslyn/pull/24469#discussion_r167677135) [](ancestors = 167677135)